modules/task-definition/container-definition/variables.tf

```diff
@@ -0,0 +1,308 @@
+  description = "The command that is passed to the sidecar container"
+  type        = list(string)
+  default     = []
+}
+
+variable "sidecar_cpu" {
+  description = "The number of cpu units used by the sidecar container"
+  type        = number
+}
+
+variable "sidecar_memory" {
+  description = "The amount of memory used by the sidecar container"
+  type        = number
+}
+
+variable "sidecar_memory_reservation" {
+  description = "The amount of memory used by the sidecar container"
+  type        = number
+}
+
+variable "sidecar_essential" {
+  description = "If the essential parameter of the sidecar container definition is marked as true"
+  type        = bool
+  default     = true
+}
+
+variable "sidecar_user" {
+  description = "The user to run the sidecar container as"
+  type        = string
+  default     = "0"
+}
+
+variable "fluentd_image" {
+  description = "The image uri used to start the sidecar container"
+  type        = string
+}
+variable "sidecar_readonly_root_filesystem" {
+  description = "If the sidecar container has a read only root filesystem"
+  type        = bool
+  default     = true
+}
+
+variable "sidecar_containerPort" {
+  description = "The port number on the sidecar container."
+  type        = number
+}
+
+variable "sidecar_hostPort" {
+  description = "The port number on the container instance to reserve for your sidecar container."
+  type        = number
+}
+
+variable "sidecar_protocol" {
+  description = "The protocol used for the sidecar container."
+  type        = string
+  default     = "tcp"
+}
+
+variable "sidecar_logDriver" {
+  description = "The log driver to use for the sidecar container."
+  type        = string
+  default     = "awslogs"
+}
+
+variable "sidecar_awslogs_create_group" {
+  description = "If the log group should be created."
+  type        = bool
+  default     = true
+}
+
+variable "sidecar_awslogs_group" {
+  description = "The log group to log to."
+  type        = string
+}
+
+variable "sidecar_awslogs_region" {
+  description = "The region of the log group."
+  type        = string
+  default     = "us-east-1"
+}
+
+variable "sidecar_awslogs_stream_prefix" {
+  description = "The prefix of the log stream."
+  type        = string
+}
+
+variable "sidecar_firelens_type" {
+  description = "The type of firelens configuration to use."
+  type        = string
+  default     = "fluentd"
+}
+
+variable "sidecar_config_file_value" {
+  description = "The value of the config file."
+  type        = string
+  default     = "/extra.conf"
+}
+
+variable "sidecar_config_file_type" {
+  description = "The type of the config file."
+  type        = string
+  default     = "file"
+}
+
+variable "container_definitions" {
+  description = "The container definitions to use for the task."
+  type = map(object(
+    {
+      name              = string
+      command           = optional(list(string))
+      cpu               = number
+      essential         = optional(bool, true)
```

Comment:
For types, you can't list the default values in the same instance. The default values are in a different attribute. Here is an example: https://github.com/mediavine/terraform-fastly-service/blob/main/variables.tf#L178

Reply:
@Keimille I guess I'm a little confused because I got it from the docs...?
https://developer.hashicorp.com/terraform/language/expressions/type-constraints#optional-object-type-attributes

Reply:
This is wild. lol. Maybe the linter is not updated then? 
![Screenshot 2024-11-07 at 12 38 06 PM](https://github.com/user-attachments/assets/152caded-2be5-4ed7-ab02-9b584bfd00f5)

Reply:
I'm going to try a newer version.

Reply:
Newer version of TF fixed this.